client_test.go

```diff
@@ -588,33 +598,22 @@ func expectRedirect(location string) func(*testing.T, *http.Response, error) {
 	}
 }
 
-//nolint:revive // linter wants ctx first, but t first is okay
-func startServer(t *testing.T, ctx context.Context, handler http.Handler) string {
+func startServer(t *testing.T, handler http.Handler) string {
 	t.Helper()
 
-	listener, err := net.Listen("tcp", "127.0.0.1:0")
-	require.NoError(t, err)
-	svr := http.Server{
+	svr := httptest.NewUnstartedServer(nil)
```

Comment:
Switch to httptest here from a linter complaining (but I forgot the error condition).